pkg/hvutils/hvutils.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package hvutils provies utility methods to access the Hivelocity API.
+package hvutils
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+)
+
+// GetInstanceTypeFromTags is a utility method to read the instance-type
+// from a slice of strings.
+// The slice is usually from the Hivelocity API of a device.
+// Example: {"instance-type=foo", "other-label"} would return "foo".
+func GetInstanceTypeFromTags(tags []string) (string, error) {
+	prefix := "instance-type="
+	instanceTypes := make([]string, 0, 1)
+	for _, tag := range tags {
+		if !strings.HasPrefix(tag, prefix) {
+			continue
+		}
+
+		instanceType := strings.TrimSpace(strings.TrimPrefix(tag, prefix))
+		instanceTypes = append(instanceTypes, instanceType)
+	}
+	if len(instanceTypes) == 0 {
+		return "", ErrNoInstanceTypeFound
+	}
+	if len(instanceTypes) > 1 {
+		return "", fmt.Errorf("%w: instanceTypes: %v",
+			ErrMoreThanOneTagFound, instanceTypes)
+	}
+	instanceType := instanceTypes[0]
+
+	if errs := validation.IsValidLabelValue(instanceType); len(errs) != 0 {
+		return "", fmt.Errorf("%w: %q %s", ErrInvalidLabelValue, instanceType,
+			strings.Join(errs, "; "))
+	}
+	return instanceType, nil
+}
+
+var (
+	ErrMoreThanOneTagFound = fmt.Errorf("more than one instance-type tag found") //nolint:revive // ...
```

Comment:
don't ignore revive

Reply:
I added comments to the code, although I think that these comments make the code harder to read. 

Now the number of characters a reader of the code has to parse with his eyes has doubled.

But I don't have a strong opinion here. If it makes the linter happy, then I am happy, too.

```
-       ErrMoreThanOneTagFound = fmt.Errorf("more than one instance-type tag found") //nolint:revive // ...
-       ErrInvalidLabelValue   = fmt.Errorf("invalid label value")                   //nolint:revive // ...
-       ErrNoInstanceTypeFound = fmt.Errorf("no instance-type tag found")            //nolint:revive // ...
+
+       // ErrMoreThanOneTagFound gets returned if more than one instance-type tag was found via the HV API.
+       ErrMoreThanOneTagFound = fmt.Errorf("more than one instance-type tag found")
+
+       // ErrInvalidLabelValue gets returned if the HV tag contains a value which is an invalid K8s label.
+       ErrInvalidLabelValue = fmt.Errorf("invalid label value")
+
+       // ErrNoInstanceTypeFound gets returned if no instance-type tag was found via the HV API.
+       ErrNoInstanceTypeFound = fmt.Errorf("no instance-type tag found")
```

Reply:
I agree but would follow the convention and the linter. It makes automated documentation via pkg.go.dev possible.